src/openlcb/DefaultNode.cxx

```diff
@@ -41,15 +41,26 @@ namespace openlcb
 
 extern void StartInitializationFlow(Node* node);
 
-DefaultNode::DefaultNode(If* iface, NodeID node_id)
+DefaultNode::DefaultNode(If* iface, NodeID node_id, bool init)
     : nodeId_(node_id), isInitialized_(0), iface_(iface)
 {
     iface_->add_local_node(this);
-    StartInitializationFlow(this);
+    if (init)
+    {
+        StartInitializationFlow(this);
+    }
 }
 
 DefaultNode::~DefaultNode()
 {
 }
 
+void DefaultNode::initialize()
```

Comment:
please add this function into DefaultTrainNode as well (in TractionTrain.hxx).
DefaultTrainNode does not inherit from defaultnode (on purpose). It does inherit from Node, so an alternative option would be to move this logic into Node.